### PR TITLE
Internal: Atomic widgets props schema default [EDS-282]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: modules/atomic-widgets/base/atomic-widget-base.php [hunk 1]
```diff
@@ -49,4 +55,37 @@ final public function get_stack( $with_common_controls = true ) {
...
+		$raw_settings = $this->get_settings();
+
+		$transformed_settings = [];
+
+		foreach ( $schema as $key => $prop ) {
+			if ( array_key_exists( $key, $raw_settings ) ) {
+				$transformed_settings[ $key ] = $raw_settings[ $key ];
+			} else {
+				$transformed_settings[ $key ] = $prop->get_default();
+			}
```
Comment (blackcolt) [inline modules/atomic-widgets/base/atomic-widget-base.php:70]: ternary?

File: modules/atomic-widgets/base/atomic-widget-base.php [hunk 2]
```diff
@@ -49,4 +55,37 @@ final public function get_stack( $with_common_controls = true ) {
...
+		$schema = static::get_props_schema();
+		$raw_settings = $this->get_settings();
+
+		$transformed_settings = [];
+
+		foreach ( $schema as $key => $prop ) {
+			if ( array_key_exists( $key, $raw_settings ) ) {
+				$transformed_settings[ $key ] = $raw_settings[ $key ];
+			} else {
+				$transformed_settings[ $key ] = $prop->get_default();
```
Comment (blackcolt) [inline modules/atomic-widgets/base/atomic-widget-base.php:69]:
```suggestion
			$transformed_settings[ $key ] = array_key_exists( $key, $raw_settings ) ?  $raw_settings[ $key ]  : $prop->get_default();
			
```

File: modules/atomic-widgets/base/atomic-widget-base.php [hunk 3]
```diff
@@ -49,4 +55,37 @@ final public function get_stack( $with_common_controls = true ) {
...
+			}
+		}
+
```
Comment (Omerisra6) [inline modules/atomic-widgets/base/atomic-widget-base.php:72]:
```suggestion
    return new Collection( $schema )->pluck( 'default' )->merge( $raw_settings )->to_array();
```

Comment (StyleShit) [reply modules/atomic-widgets/base/atomic-widget-base.php:72]: Do you want Rotem to kill me?

Comment (Omerisra6) [reply modules/atomic-widgets/base/atomic-widget-base.php:72]: you can decode it first....
But just a recommendation

File: modules/atomic-widgets/base/atomic-widget-base.php [hunk 4]
```diff
@@ -49,4 +55,37 @@ final public function get_stack( $with_common_controls = true ) {
...
+	public static function get_props_schema(): array {
+		$cache_key = static::class;
+
+		if ( empty( static::$props_schema_cache[ $cache_key ] ) ) {
+			static::$props_schema_cache[ $cache_key ] = static::get_props_schema_definition();
+		}
+
+		return static::$props_schema_cache[ $cache_key ];
+	}
```
Comment (Nevoss) [inline modules/atomic-widgets/base/atomic-widget-base.php:85]: Check if it really improve performance

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: modules/atomic-widgets/base/atomic-widget-base.php (hunk 4 above)
Comment (StyleShit) [reply modules/atomic-widgets/base/atomic-widget-base.php:85]: I'm just removing it until we get to a point we need to optimize it...

**Review by Nevoss — APPROVED** [review]

**Pull request merged**